### PR TITLE
ECONOMY SYSTEM

### PR DESCRIPTION
**Pull request by JordiHakimZiyech**
Joe please take a look and leave suggestions where needed.
2 more commands will be changed from file soon to this economy file (conflip and slotmachine)

### CODE REVIEW

File: economy/roll.go.tmpl [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	Trigger type: Command
+ 
+	Credits:
+	Made by Jordi
+*/}}
+{{/* Configuration variables start */}}
+{{ $name := "coins" }} {{/* the name of the currency */}}
+{{ $bet := 75 }} {{/* Amount to bet */}}
+{{/* Configuration variables end */}}
+{{ $newcoins := sub ((dbGet .User.ID "coins").Value) 75 }}
```
Comment (DaviiD1337) [inline economy/roll.go.tmpl:16]: {{ $newcoins := sub ((dbGet .User.ID "coins").Value) $bet }} 👍

File: economy/work.go.tmpl [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	Made by Jordi
+*/}}
+{{/* Configuration variables start */}}
+{{ $name := "coins" }} {{/* the name of the currency */}}
+{{ $min := 500 }} {{/* the minimum amount people should get for working*/}}
+{{ $max := 1000}} {{/* the maximum amount people should get for working*/}}
+{{ $cooldown := 20}} {{/* the cooldown before users can work again*/}}
+{{/* Configuration variables end */}}
+
+{{ $subt := randInt $min $max }}
```
Comment (DaviiD1337) [inline economy/work.go.tmpl:18]: maybe check if min < max so it wont give error

File: economy/rps.go.tmpl [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	Made by Jordi
+*/}}
+{{/* Configuration variables start */}}
+{{ $name := "coins" }} {{/* the name of the currency */}}
+{{ $bet := 75 }} {{/* amount to bet */}}
+{{ $cooldown := 120 }} {{/* cooldown in seconds */}}
+{{/* Configuration variables end */}}
+
+{{ $var := randInt 3 }}
+{{ $newcoins := sub ((dbGet .User.ID "coins").Value) $bet }}
```
Comment (DaviiD1337) [inline economy/rps.go.tmpl:18]: {{ $newcoins := sub ((dbGet .User.ID $name).Value) $bet }}

File: economy/rps.go.tmpl [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+ 
+{{ $args := parseArgs 1 "usage is: -rps [rock/paper/scissor]"
+(carg "string" "choice") }}
+{{ if (dbGet .User.ID "rpscooldown") }}
+{{ sendMessage nil (cembed "author" (sdict "name" .User.String "icon_url" (.User.AvatarURL "512")) "title" "RPS" "color" 16647943 "description" (print "You have to wait " (humanizeDurationSeconds ((dbGet .User.ID "rpscooldown").ExpiresAt.Sub currentTime)) " before you can play again")) }} {{ else }} {{ dbSetExpire .User.ID "rpscooldown" 1 $cooldown }}
+{{ if lt (toInt (dbGet .User.ID "coins").Value) $bet }}
```
Comment (DaviiD1337) [inline economy/rps.go.tmpl:24]: same as up ^
Also maybe check if inFold (cslice blablba) the chosen thing.

File: economy/stealDepWith.go.tmpl [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+ 
+	Credits:
+	Made by Jordi
+{{/* Configuration variables start */}} 
+{{ $name := "coins" }} {{/* the name of the currency */}}
+{{ $cooldown := 21600 }} {{/* Cooldown in seconds before someone can rob/get robbed */}}
+{{/* Configuration variables end */}}
+{{if reFindAllSubmatches `\A(-|<@!?204255221017214977>\s*)(rob|steal)(\s+|\z)` .Message.Content}}
+{{ $args := parseArgs 1 "-steal user" (carg "userid" "user") }}
+{{ $lost1W := mult 0.1 (toInt (dbGet ($args.Get 0) "coins").Value) }}
```
Comment (DaviiD1337) [inline economy/stealDepWith.go.tmpl:18]: dbGet $args.get 0 $name again

File: economy/stealDepWith.go.tmpl [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+{{ $lostW := sub (toInt (dbGet ($args.Get 0) "coins").Value) (toInt $lost1W) }}
```
Comment (DaviiD1337) [inline economy/stealDepWith.go.tmpl:19]: again ^

File: economy/stealDepWith.go.tmpl [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+{{ else }} {{ $frick := dbIncr .User.ID "coins" $lost1W }} {{ sendMessage nil (cembed "author" (sdict "name" .User.String "icon_url" (.User.AvatarURL "512")) "title" "Steal" "color" 3014151 "description" (print "You robbed " (userArg ($args.Get 0)).Mention " and got " (humanizeThousands $lost1W) " " $name)) }} {{ dbSet ($args.Get 0) "coins" $lostW }} {{ end }} {{ end }}
+{{else if reFindAllSubmatches `\A(-|<@!?204255221017214977>\s*)(with|withdraw)(\s+|\z)` .Message.Content}}
+{{ $args := parseArgs 1 "usage is: -withdraw amount" (carg "int" "amount of coins") }}
+{{ if ge ($args.Get 0) 1 }}
+{{ if ge (toInt (dbGet .User.ID "bank").Value) ($args.Get 0) }}
+{{ sendMessage nil (cembed "author" (sdict "name" .User.String "icon_url" (.User.AvatarURL "512")) "title" "Deposit" "color" 3014151 "description" (print .User " has withdrawn " ($args.Get 0) " " $name)) }} {{ $up := dbIncr .User.ID "coins" ($args.Get 0) }} 
+{{ $newcoins := sub (toInt (dbGet .User.ID "bank").Value) ($args.Get 0) }}
```
Comment (DaviiD1337) [inline economy/stealDepWith.go.tmpl:30]: what is db named "bank"

File: economy/setCurrency.go.tmpl [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+	Credits:
+	Made by Jordii
+*/}}
+{{/* Configuration variables start */}}
+{{ $name := "coins" }} {{/* the name of the currency */}}
+{{/* Configuration variables end */}}
+{{ $args := parseArgs 2 "usage is: -set @user amount"
+(carg "userid" "user to give currency to")
+    (carg "int" "amount of currency") }}
+{{ dbSet ($args.Get 0) "coins" ($args.Get 1) }}
```
Comment (DaviiD1337) [inline economy/setCurrency.go.tmpl:18]: again $name on dbset

File: economy/giveCurrency.go.tmpl [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+	Made by Jordi
+*/}}
+{{/* Configuration variables start */}}
+{{ $name := "coins" }} {{/* the name of the currency */}}
+{{/* Configuration variables end */}}
+{{ $args := parseArgs 2 "usage is: -give @user amount"
+(carg "userid" "user to give currency to")
+    (carg "int" "amount of currency") }}
+{{ if gt ($args.Get 1) 1 }}
+{{ if gt (toInt (dbGet .User.ID "coins").Value) ($args.Get 1) }}
```
Comment (DaviiD1337) [inline economy/giveCurrency.go.tmpl:18]: $name

**Review by DaviiD1337 — CHANGES_REQUESTED** [review]

File: economy/leaderboard/lb.go.tmpl [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+	Credits:
+	Made by Joe, edited by Jordi
+*/}}
+{{/* Configuration variables start */}}
+{{ $name := "coins" }} {{/* the name of the currency */}}
+{{/* Configuration variables end */}}
+{{ $page := 1 }} {{/* Default page to start at */}}
+{{ with reFind `\d+` (joinStr " " .CmdArgs) }} {{ $page = . | toInt }} {{ end }} {{/* If the user provided a page, change $page variable to that */}}
+{{ $skip := mult (sub $page 1) 10 }} {{/* Amount of entries to skip */}}
+{{ $users := dbTopEntries "coins" 10 $skip }} {{/* Retrieve the relevant DB entries with the parameters provided */}}
```
Comment (DaviiD1337) [inline economy/leaderboard/lb.go.tmpl:17]: $name

File: economy/leaderboard/reaction.go.tmpl [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+		{{ $isValid = true }} {{/* Yay, it is valid */}}
+		{{ deleteMessageReaction nil $.ReactionMessage.ID $.User.ID $action }}
+	{{ end }}
+{{ end }}
+{{ if and (in $validEmojis $action) $isValid $page }} {{/* Even more checks for validity... */}}
+	{{ if eq $action "👉" }} {{ $page = add $page 1 }} {{/* Update page according to emoji */}}
+	{{ else if eq $action "👈"}} {{ $page = sub $page 1 }} {{ end }}
+	{{ if ge $page 1 }} {{/* Otherwise, dbTopEntries throws error due to negative skip */}}
+		{{ $skip := mult (sub $page 1) 10 }} {{/* Get skip */}}
+		{{ $users := dbTopEntries "coins" 10 $skip }} {{/* Fetch entries */}}
```
Comment (DaviiD1337) [inline economy/leaderboard/reaction.go.tmpl:30]: $name

**Review by DaviiD1337 — CHANGES_REQUESTED** [review]

File: economy/stealDepWith.go.tmpl (hunk 7 above)
Comment (JordiHakimZiyech) [reply economy/stealDepWith.go.tmpl:30]: the bank, where u store ur currency

File: economy/roll.go.tmpl (hunk 1 above)
Comment (JordiHakimZiyech) [reply economy/roll.go.tmpl:16]: ow shit ty

File: economy/work.go.tmpl (hunk 2 above)
Comment (JordiHakimZiyech) [reply economy/work.go.tmpl:18]: will do

File: economy/rps.go.tmpl (hunk 3 above)
Comment (JordiHakimZiyech) [reply economy/rps.go.tmpl:18]: no no no, the complete system is based on db entry "coins" for now, ill keep it like that, just to prevent issues

**Comment by JordiHakimZiyech** [comment]
closing for now, after discussion with Joe.

**Pull request closed without merging**